app/src/main/java/com/losandroides/learn/di/MainProvidesModule.kt

```diff
@@ -14,6 +22,24 @@ class MainProvidesModule {
     @Provides
     fun provideRetrofitClient(): RetrofitClient = RetrofitClient()
 
+    @Provides
+    fun provideItemDatabase(app: Application): ItemDatabase = Room.databaseBuilder(
+        app,
+        ItemDatabase::class.java,
+        "item-db"
+    ).build()
+
     @Provides
     fun provideItemService(retrofitClient: RetrofitClient): ItemService = retrofitClient.itemService
+
+    @Provides
+    fun provideLocalPointsDataSource(database: ItemDatabase): LocalItemDatasource =
+        DefaultLocalItemDatasource(
+            database,
+            Dispatchers.IO
```

Comment:
Do you know how could we provide this `Dispatchers.IO` getting the one in the `CoroutinesDispatchersModule`, @RubenGonz2?

Reply:
Sorry Gabi. I don't use Dagger usually. I usually use a dependency injection patter implemented by my own. So I don't know how to do it. I want to test Dagger hilt in a small project I'm doing. So if I learn how to do it I'll come back and try to implement it in this project.